test/uk/gov/hmrc/merchandiseinbaggage/pagespecs/pages/CheckYourAnswersPage.scala

```diff
@@ -85,8 +28,13 @@ object CheckYourAnswersPage extends Page {
   val expectedSectionHeaders =
     Seq("Details of the goods", "Personal details", "Journey details", "Now send your declaration")
 
-  def submitPage()(implicit webDriver: HtmlUnitDriver): Unit = {
-    val button = find(NameQuery("makeDeclarationButton")).get
+  def submitPage[T](formData: T)(implicit webDriver: HtmlUnitDriver): Unit = {
+    val dType = formData.asInstanceOf[DeclarationType]
```

Comment:
Do you need to use   asInstanceOf, could it be...
```
  val query = formData match {
      case Export => "makeDeclarationButton"
      case Import => "payButton"
      case _      => fail("INVALID TYPE")
    }
```

Reply:
hmm, ok, i can fix it in the upcoming PR part 3 if thats ok